File: Exercise 01 - Check sum == k/C++ code/Exercise1.cpp
/*Given a list of numbers and a number k, return whether any two numbers from the list add up to k.
For example, given[10, 15, 3, 7] and k of 17, return true since 10 + 7 is 17.*/

#include <iostream>
#include <vector>
#include <algorithm>

template <typename T>
auto f(const std::vector<T>& vec, T index)
{
    for (auto i = vec.begin(); i != vec.end(); i++)
    {
        auto k = index - (*i);
        auto j = i;
        if (std::find(++j, end(vec), k) != end(vec))
        {
            return true;
        }
    }
    return false;
}


int main()
{
    std::vector<double> vec{ 10,15,3,7 };
    auto index{ 17. };
    auto index2{ 19. };
    std::cout << f(vec, index) << std::endl;
    std::cout << f(vec, index2) << std::endl;
}

File: Daily Coding problems/Exercise 01 - Check sum == k/Python code/Exercise1.py
"""Given a list of numbers and a number k, return whether any two numbers from the list add up to k.
For example, given[10, 15, 3, 7] and k of 17, return true since 10 + 7 is 17."""

import numpy as np

def f(vec, index):
    for idx,i in enumerate(vec):
        k = index - i;
        j = idx+1;
        if (k in vec[j:]):
            return True
    return False

vec = np.array([10,15,3,7])
index = 17.
index2 = 19.
print(f(vec, index))
print(f(vec, index2))

